### ISSUE TITLE
@bugsnag/node bundle size 

### ISSUE THREAD

**Issue by kwonoj** [bugsnag/bugsnag-js#456]
![image](https://user-images.githubusercontent.com/1210596/50358108-1960d180-050d-11e9-9552-666f679df090.png)

Check above for reference. Our main process's bundle size in total is about 1.4MB, `node_modules` takes around 1.1 and @bugsnag/node takes 360KB, nearly 1/4 of whole bytesize. As an error event reporter, that is non trivial size and we are blocked to optimize our bundle size - is there any viable approach to reduce bundle sizes?

**Comment by snmaynard** [comment]
We have an open ticket internally to move from request to something like https-proxy-agent which should reduce the bundle size by quite a bit.

Here is the dependency tree - so moving from request will knock a few of those off. We could also look into using a plugin based approach to add proxy support if it increases the bundle size too much and people can opt into it.

```
/Users/simon/Projects/test
└─┬ @bugsnag/node@5.1.0
  ├── byline@5.0.0
  ├─┬ error-stack-parser@2.0.2
  │ └── stackframe@1.0.4
  ├── iserror@0.0.2
  ├─┬ pump@3.0.0
  │ ├─┬ end-of-stream@1.4.1
  │ │ └── once@1.4.0 deduped
  │ └─┬ once@1.4.0
  │   └── wrappy@1.0.2
  ├─┬ request@2.88.0
  │ ├── aws-sign2@0.7.0
  │ ├── aws4@1.8.0
  │ ├── caseless@0.12.0
  │ ├─┬ combined-stream@1.0.7
  │ │ └── delayed-stream@1.0.0
  │ ├── extend@3.0.2
  │ ├── forever-agent@0.6.1
  │ ├─┬ form-data@2.3.3
  │ │ ├── asynckit@0.4.0
  │ │ ├── combined-stream@1.0.7 deduped
  │ │ └── mime-types@2.1.21 deduped
  │ ├─┬ har-validator@5.1.3
  │ │ ├─┬ ajv@6.6.2
  │ │ │ ├── fast-deep-equal@2.0.1
  │ │ │ ├── fast-json-stable-stringify@2.0.0
  │ │ │ ├── json-schema-traverse@0.4.1
  │ │ │ └─┬ uri-js@4.2.2
  │ │ │   └── punycode@2.1.1
  │ │ └── har-schema@2.0.0
  │ ├─┬ http-signature@1.2.0
  │ │ ├── assert-plus@1.0.0
  │ │ ├─┬ jsprim@1.4.1
  │ │ │ ├── assert-plus@1.0.0 deduped
  │ │ │ ├── extsprintf@1.3.0
  │ │ │ ├── json-schema@0.2.3
  │ │ │ └─┬ verror@1.10.0
  │ │ │   ├── assert-plus@1.0.0 deduped
  │ │ │   ├── core-util-is@1.0.2
  │ │ │   └── extsprintf@1.3.0 deduped
  │ │ └─┬ sshpk@1.16.0
  │ │   ├─┬ asn1@0.2.4
  │ │   │ └── safer-buffer@2.1.2 deduped
  │ │   ├── assert-plus@1.0.0 deduped
  │ │   ├─┬ bcrypt-pbkdf@1.0.2
  │ │   │ └── tweetnacl@0.14.5 deduped
  │ │   ├─┬ dashdash@1.14.1
  │ │   │ └── assert-plus@1.0.0 deduped
  │ │   ├─┬ ecc-jsbn@0.1.2
  │ │   │ ├── jsbn@0.1.1 deduped
  │ │   │ └── safer-buffer@2.1.2 deduped
  │ │   ├─┬ getpass@0.1.7
  │ │   │ └── assert-plus@1.0.0 deduped
  │ │   ├── jsbn@0.1.1
  │ │   ├── safer-buffer@2.1.2
  │ │   └── tweetnacl@0.14.5
  │ ├── is-typedarray@1.0.0
  │ ├── isstream@0.1.2
  │ ├── json-stringify-safe@5.0.1
  │ ├─┬ mime-types@2.1.21
  │ │ └── mime-db@1.37.0
  │ ├── oauth-sign@0.9.0
  │ ├── performance-now@2.1.0
  │ ├── qs@6.5.2
  │ ├── safe-buffer@5.1.2
  │ ├─┬ tough-cookie@2.4.3
  │ │ ├── psl@1.1.31
  │ │ └── punycode@1.4.1 extraneous
  │ ├─┬ tunnel-agent@0.6.0
  │ │ └── safe-buffer@5.1.2 deduped
  │ └── uuid@3.3.2
  └─┬ stack-generator@2.0.3
    └── stackframe@1.0.4 deduped
```

**Comment by kwonoj** [comment]
Thanks, glad to hear it's in consideration.

**Comment by kwonoj** [comment]
(Please feel free tonclose if there's dupe issue already)

**Comment by kwonoj** [comment]
@snmaynard Please allow me one additional question : is there plan to support electron specific path, i.e use `electron.net` module for network request uses out of box support from framework?

**Comment by snmaynard** [comment]
We are discussing a native electron notifier internally - it might make sense for us to reach out to you to talk about it when the time comes if you are open to that?

**Comment by kwonoj** [comment]
Sure.

**Comment by bengourley** [comment]
[`request`](https://www.npmjs.com/package/request) was removed and replaced with a node-core delivery mechanism in #478. According to [bundlephobia](https://bundlephobia.com/result?p=@bugsnag/node@6.0.0-alpha.1) the size went from around 700kB to 40kB when comparing the current stable version (`v5.2.0`) to a prerelease including this change (`v6.0.0-alpha.1`).

This update will be released as a major version `v6.0.0` in the next few weeks.
